src/iris/bin/app_stats.py

```diff
@@ -44,22 +45,30 @@ def set_global_stats(stats, connection, cursor):
     # delete outdated stats
     cursor.execute('''DELETE FROM `global_stats`''')
 
-    for stat in stats.keys():
-        for timestamp, val in stats[stat].items():
-            cursor.execute('''INSERT INTO `global_stats` (`statistic`, `value`, `timestamp`)
-                            VALUES (%s, %s, %s)
-                            ON DUPLICATE KEY UPDATE `value`= %s, `timestamp` = %s''',
-                           (stat, val, timestamp, val, timestamp))
+    # format: {statistic : [{timestamp: value}, {timestamp: value}]}
+    for stat, entry_list in stats.items():
+        for stat_entry in entry_list:
+            for timestamp, val in stat_entry.items():
+                # format timestamp into datetime to store in mysql
+                timestamp = datetime.utcfromtimestamp(timestamp).strftime('%Y-%m-%d %H:%M:%S')
```

Comment:
For future reference, `FROM_UNIXTIME()` will do this in SQL, so you don't have to mess with datetime and timestring formats.